src/client/mod.rs

```diff
@@ -114,6 +114,11 @@ impl MsgEnvelope {
         Ok(sender.verify(&data))
     }
 
+    /// Get Message target section's expected PublicKey
+    pub fn target_section_pk(&self) -> Option<PublicKey> {
+        self.message.target_section_pk()
```

Comment:
Why is the target section pk optional?

Reply:
I guess this could be optional for an infrastructure query, but is it optional for all other type of messages?

Reply:
It's not always required as far as I understood.

Reply:
right, so i guess this function shall return the option then, but i guess moving forward we can make it non-optional for the messages types which always require it, I'm now talking about the actual enums not this function.